### PR TITLE
[M] CANDLEPIN-122: Move Jenkins PR mariadb pipeline stages to use Centos Stream 8

### PR DESCRIPTION
**Pull request by bakajstep**
MariaDB pipeline stages are set to run on Centos Stream 8.
Adding support to run MariaDB stages on different OS.

### CODE REVIEW

File: docker/docker-compose-mysql-cs8.yml [hunk 1]
```diff
@@ -8,7 +8,7 @@ services:
     volumes:
         - ./mysql.cnf:/etc/mysql/conf.d/mysql.cnf
   candlepin:
-    image: ${REGISTRY}/candlepin-base
+    image: ${REGISTRY}/candlepin-base-cs8
```
Comment (nikosmoum) [inline docker/docker-compose-mysql-cs8.yml:11]: Wouldn't this change be enough to make this work? Why the rest?

File: docker/test [hunk 2]
```diff
@@ -11,7 +11,7 @@ USAGE: test [OPTIONS]
 OPTIONS:
   -c CMD    sets the command to be run by the container
               default: $TEST_CMD
-  -m        use mysql
+  -m [OS]   use mysql, we can specify OS where candlepin will be running. Default is Centos Stream 8: cs8.
```
Comment (nikosmoum) [inline docker/test:14]: If we need to specify the OS/image version, It would be much better to use a completely new argument (e.g. `-a [image]` instead of making this an argument on the mysql flag

File: jenkins/Jenkinsfile [hunk 3]
```diff
@@ -108,6 +108,7 @@ pipeline {
                     environment {
                         CANDLEPIN_DATABASE = 'mysql'
                         CP_TEST_ARGS = '-r'
+                        OS_IMAGE = 'cs8'
```
Comment (nikosmoum) [inline jenkins/Jenkinsfile:111]: why is this only set at the standalone-mysql? should it also not be on the hosted-mysql stage?

**Review by nikosmoum — CHANGES_REQUESTED** [review]

File: jenkins/Jenkinsfile (hunk 3 above)
Comment (bakajstep) [reply jenkins/Jenkinsfile:111]: I want to just show how you can specify that OS, so i just set it to one job. So next person can see you could select OS or not and use default.

File: docker/docker-compose-mysql-cs8.yml (hunk 1 above)
Comment (bakajstep) [reply docker/docker-compose-mysql-cs8.yml:11]: yes, but i was thinking if we decide in future to run different OS like with postgres, so I thought it would be ready.

File: docker/test (hunk 2 above)
Comment (bakajstep) [reply docker/test:14]: I know we talked about it, but then I thought about it and I thought that there are a lot of edge cases that we'll have to cover(both parameters need to be set, different OS for mariadb and postgres and more ) and I don't know who all uses the script, so this seemed like a safe choice, but I can redo it if you want.

Comment (nikosmoum) [reply docker/test:14]: I think we should re-do it by using a separate flag. This is not intuitive and will confuse people, because one flag is used for specifying two arguments (db, and image)

File: jenkins/Jenkinsfile (hunk 3 above)
Comment (nikosmoum) [reply jenkins/Jenkinsfile:111]: If it is not functional, we should remove it. Anyone looking to do something like this will be able to tell by checking `docker/test --help`

File: docker/test (hunk 2 above)
Comment (bakajstep) [reply docker/test]: done

File: jenkins/Jenkinsfile (hunk 3 above)
Comment (bakajstep) [reply jenkins/Jenkinsfile:111]: it has functionality, but same as default. Ok, I remove it.

File: docker/test [hunk 4]
```diff
@@ -2,6 +2,7 @@
 
 TEST_CMD="/usr/bin/cp-test -t -u -r"
 DIR="$(git rev-parse --show-toplevel)/docker/"
+OPERATION_SYSTEM="cs8" # Changing this variable, change default OS
```
Comment (nikosmoum) [inline docker/test:5]: nitpick: OPERATING instead of OPERATION

File: docker/test [hunk 5]
```diff
@@ -11,6 +12,15 @@ USAGE: test [OPTIONS]
 OPTIONS:
   -c CMD    sets the command to be run by the container
               default: $TEST_CMD
+  -o OS     Specify operation system where candlepin will be running
```
Comment (nikosmoum) [inline docker/test:15]: same here: operating

File: docker/candlepin-base-cs8/setup-devel-env.sh [hunk 6]
```diff
@@ -81,6 +81,9 @@ bundle install --without=proton --retry=5 --verbose --full-index
 # Installs all Java deps into the image, big time saver
 ./gradlew --no-daemon dependencies
 
+# Fix issue with owner on mount volume
```
Comment (nikosmoum) [inline docker/candlepin-base-cs8/setup-devel-env.sh:84]: Does this issue happen only on cs8 and cs9? Or does it also need to be fixed in c7 (because older branches still need that)?

File: docker/test [hunk 7]
```diff
@@ -54,6 +66,15 @@ while getopts ":c:mn:pxqlde" opt; do
   esac
 done
 
+COMPOSE_ARGS="$COMPOSE_ARGS-$OPERATION_SYSTEM.yml"
+
+if ! [ $COMPOSE_ARGS ];
+then
+  echo "Docker compose file does not exist"
```
Comment (nikosmoum) [inline docker/test:73]: Since this error is the result of someone choosing an operating system that isn't supported, like `-o cs10`, can we change this to something like `Operating system not supported`?

**Review by nikosmoum — CHANGES_REQUESTED** [review]

File: docker/test (hunk 5 above)
Comment (bakajstep) [reply docker/test]: done

File: docker/test (hunk 7 above)
Comment (bakajstep) [reply docker/test]: Done, I also print that bad os what user enter.

**Review by nikosmoum — APPROVED** [review]
